In [1]:
import pandas as pd
from scipy import stats
import numpy as np


# Descriptive analytics

In [7]:
df=pd.read_csv('Week_8_Q&A_dataset - Sheet1.csv')
print(df)

    Gender Test preparation  Total Marks
0     male             none           14
1   female             none           28
2   female             none           18
3   female             none           48
4   female             none           21
5   female        completed           40
6     male             none           30
7   female             none           12
8     male             none           18
9     male             none           24
10    male        completed           44
11    male             none           23
12    male        completed           43
13    male             none           28
14  female             none           41
15  female             none           13
16  female        completed           48
17  female             none           45
18    male             none           47
19    male             none           36
20    male             none           14
21    male        completed           44
22  female             none           41
23  female      

In [8]:
df.head()

Gender Test preparation  Total Marks
0    male             none           14
1  female             none           28
2  female             none           18
3  female             none           48
4  female             none           21

In [9]:
df['Gender'].value_counts()

female    16
male      12
Name: Gender, dtype: int64

In [10]:
df.describe()

Total Marks
count    28.000000
mean     32.321429
std      12.454520
min      12.000000
25%      22.500000
50%      33.000000
75%      43.250000
max      50.000000

In [12]:
df.isna().sum()

Gender              0
Test preparation    0
Total Marks         0
dtype: int64

In [13]:
df.groupby(['Gender']).mean()

Total Marks
Gender             
female    33.750000
male      30.416667

In [14]:
df.groupby(['Test preparation']).mean()

Total Marks
Test preparation             
completed           44.571429
none                28.238095

# Check whether female and male students scored the same marks.


Null hypothesis H0: male marks < female marks

Alternate Hypothesis H1: male marks > female marks

In [15]:
boys_mark=df[df['Gender']=='male']
boys_mark['Total Marks']

0     14
6     30
8     18
9     24
10    44
11    23
12    43
13    28
18    47
19    36
20    14
21    44
Name: Total Marks, dtype: int64

In [16]:
female_mark=df.loc[(df['Gender']=='female')]
female_mark['Total Marks']


1     28
2     18
3     48
4     21
5     40
7     12
14    41
15    13
16    48
17    45
22    41
23    27
24    27
25    38
26    50
27    43
Name: Total Marks, dtype: int64

In [17]:
t_stat,p_value=stats.ttest_ind(boys_mark['Total Marks'],female_mark['Total Marks'],alternative='less')

In [18]:
print('T Statistics value is: ',t_stat, 'P value is: ',p_value)


T Statistics value is:  -0.6940885268848668 P value is:  0.24689111589803814


T Statistics value is:  -0.6940885268848668 P value is:  0.24689111589803814
Let us assume that alpha value is 0.05, our p value is 0.246 which is greater than alpha value. So, we reject null hypothesis.

Conclusion:- Female students scored higher than male

# Whether test preparation helps the students?

H0: the students who completed the test preparation <= students who didn't completed the test preparation

H1: the students who completed the test preparation > students who didn't completed the test preparation




In [19]:
completed_marks=df.loc[(df['Test preparation']=='completed')]

In [20]:
completed_marks['Total Marks']

5     40
10    44
12    43
16    48
21    44
26    50
27    43
Name: Total Marks, dtype: int64

In [22]:
none_marks=df.loc[(df['Test preparation']=='none')]


In [23]:
none_marks['Total Marks']

0     14
1     28
2     18
3     48
4     21
6     30
7     12
8     18
9     24
11    23
13    28
14    41
15    13
17    45
18    47
19    36
20    14
22    41
23    27
24    27
25    38
Name: Total Marks, dtype: int64

In [24]:
t_stat,P_value=stats.ttest_ind(completed_marks['Total Marks'],none_marks['Total Marks'],alternative='greater')
print('T-statistics value is: ',t_stat,'P-Value is: ',P_value)

T-statistics value is:  3.6143637527769217 P-Value is:  0.0006335646228660053


T-statistics value is:  3.6143637527769217 P-Value is:  0.0006335646228660053
Let us assume that our alpha is 0.05, from the above we can see that our value for P-value is 0.0006. So, we reject the null hypothesis value.

Conclusion:-The students who completed test preparation scored greater than the students who didn't.